libraries/AC_AutoTune/AC_AutoTune_FreqResp.cpp

```diff
@@ -281,19 +285,29 @@ void AC_AutoTune_FreqResp::push_to_tgt_buffer(uint16_t count, float amplitude, u
     sample.curr_count = count;
     sample.amplitude = amplitude;
     sample.time_ms = time_ms;
-    tgt_peak_info_buffer.push(sample);
+    tgt_peak_info_buffer->push(sample);
 
 }
 
 // pull target peak info from buffer
 void AC_AutoTune_FreqResp::pull_from_tgt_buffer(uint16_t &count, float &amplitude, uint32_t &time_ms)
 {
     peak_info sample;
-    if (!tgt_peak_info_buffer.pop(sample)) {
+    if (!tgt_peak_info_buffer->pop(sample)) {
         // no sample
         return;
     }
     count = sample.curr_count;
     amplitude = sample.amplitude;
     time_ms = sample.time_ms;
 }
+
+void AC_AutoTune_FreqResp::set_dwell_cycles(uint8_t cycles)
+{
+    dwell_cycles = cycles;
+    if (meas_peak_info_buffer != nullptr) { delete meas_peak_info_buffer;}
+    meas_peak_info_buffer = new ObjectBuffer<peak_info>(cycles);
```

Comment:
Need to used the new `new` macro.
```suggestion
    meas_peak_info_buffer = NEW_NOTHROW ObjectBuffer<peak_info>(cycles);
```